File: Python/day1/main.py
with open('input.txt') as file:
    lines = [l.split() for l in file.read().split('\n')]
#print(lines)


line1 = []; line2 = []
for l in lines:
    line1.append(int(l[0]))
    line2.append(int(l[1]))

#print(line1, line2)

line1.sort()
line2.sort()

#print(line1, line2)

total = 0
for i, j in zip(line1, line2):
    total += abs(j-i)

print('Part 1:', total)
    
###Part 2 

total = 0
for i in line1:
    total += (line2.count(i) * i)

print('Part 2:', total)

File: Java/day1/src/Main.java
import java.util.ArrayList;
import java.util.Collections;

public class Main {
    
    public static void main(String[] args) {
        FileOps fileOps = new FileOps();

        ArrayList<String> data = fileOps.read("day1\\src\\input.txt");
        //System.out.println(data);

        ArrayList<Integer> line1 = new ArrayList<>();
        ArrayList<Integer> line2 = new ArrayList<>();

        for (String i : data) {
            String[] temp = i.split("   ");
            line1.add(Integer.parseInt(temp[0]));
            line2.add(Integer.parseInt(temp[1]));
        }
        line1.sort(null);
        line2.sort(null);


        Integer total = 0;
        for (int i = 0; i < line1.size(); i++) {
            Integer d = line1.get(i) - line2.get(i);
            if (d < 0) {
                total -= d;
            } else {
                total += d;
            }
        }

        System.out.println("Part 1: " + total);
        
        //Part 2
        total = 0;
        for (Integer i : line1) {
            total += i * Collections.frequency(line2, i);
        }

        System.out.println("Part 2: " + total);

    }

}

